java-frontend/src/test/java/org/sonar/java/model/JParserSemanticTest.java

```diff
@@ -1207,6 +1208,30 @@ void declaration_method_parameterized_recovered() {
     assertThat(metadata.annotations()).isEmpty();
   }
 
+  @Test
+  void ecj_exception_when_computing_metadata_should_be_caught() {
+    String source = "" +
+      " public class C {\n" +
+      "  interface I1 {}\n" +
+      "  interface I2 {\n" +
+      "    @I1(\"\")\n" +
```

Comment:
worth adding a comment that it explicitly does not compile, as `I1` is not an `@interface`.